### DISCUSSION TITLE
Github VS code failed to connect

### DISCUSSION THREAD

**Discussion by 0xkriswebchain** [Cyfrin/foundry-full-course-cu#1398] [General]
When use command: git push -u origin main
I get this issue
ssh: connect to host ssh.github.com port 443: Connection timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

Please help to resolve this issue

**Comment by Tusharkuma** [comment]
put a command 'git remote -v' in your terminal, to see if your git is connected to your remote repo.

**Reply by 0xkriswebchain** [reply]
I tried with this command the output is nothing. i tried to command such as forge init, forge compile, build and other command as well.  but i get this error "fatal: unable to access 'https://github.com/foundry-rs/forge-std/': Failed to connect to github.com port 443 after 130786 ms: Connection timed out
fatal: clone of 'https://github.com/foundry-rs/forge-std' into submodule path '/home/kriswebchain/foundry-dog-nft/lib/forge-std' failed
Error:"

**Reply by Tusharkuma** [reply]
the PROBLEM is not with foundry, so no need for foundry's command. The problem is that your git is not connected to remote github repo.
So, first initialize git onto your project and connect it with your github repo. Use command ' git init '  to initialize, 
then ' $ git config --global user.name "John Doe"
$ git config --global user.email 'johndoe@example.com ' to config your github to git. 
                                                              OR
follow the instructions from the link - https://youtu.be/jT9QEUOX00w?si=mO5GvMh1DOYMHvRs 

let me know if everything worked.

**Comment by 0xkriswebchain** [comment]
The issue is resolved once i downloaded the git desktop version. Thank you for your time and concern.
